packages/flutter_test/lib/src/binding.dart

```diff
@@ -611,6 +624,20 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
   late StackTraceDemangler _oldStackTraceDemangler;
   FlutterErrorDetails? _pendingExceptionDetails;
 
+  MockMessageHandler? _announcementHandler;
+  List<Announcement>? _announcements;
+
+  /// Returns a list announcements made by the Flutter framework.
+  List<Announcement>? takeAnnouncements() {
+    assert(inTest);
+    return _announcements;
+  }
+
+  /// Returns the most recent announcement made by the Flutter framework.
+  Announcement? getLastAnnouncement(){
```

Comment:
If this is meant to return the last announcement without removing it, I'd call it `peekLastAnnouncement` rather than `get`. But I think perhaps we want `take` here as well, no?

Reply:
So `take` would remove the announcement from the list? In that case if the user calls `takeLastAnnouncement` twice, the second time what they'll get is not actually the last announcement. I'm not sure if this can cause confusion or if it's a common practice.

Reply:
Ah, good point. In that case maybe it should be called just `takeAnnouncement`.

Reply:
If I remember correctly, we decided that we don't need both `takeAnnouncement` and `takeAnnouncements`.
I've also renamed `getLastAnnouncement` to `peekLastAnnouncement`.

Reply:
Based on the other comments, I'm removing this function all together because it's easy to access the last announcement through `takeAnnouncements()` and we don't really need a separate function.